alert-common/src/main/java/com/synopsys/integration/alert/common/persistence/model/job/DistributionJobModel.java

```diff
@@ -0,0 +1,164 @@
+import java.util.UUID;
+
+import org.jetbrains.annotations.Nullable;
+
+import com.synopsys.integration.alert.common.enumeration.FrequencyType;
+import com.synopsys.integration.alert.common.enumeration.ProcessingType;
+import com.synopsys.integration.alert.common.persistence.model.job.details.DistributionJobDetailsModel;
+import com.synopsys.integration.alert.common.rest.model.AlertSerializableModel;
+
+public class DistributionJobModel extends AlertSerializableModel {
+    private final UUID jobId;
+    private final boolean enabled;
+    private final String name;
+    private final FrequencyType distributionFrequency;
+    private final ProcessingType processingType;
+    private final String channelDescriptorName;
+    private final OffsetDateTime createdAt;
+    private final OffsetDateTime lastUpdated;
+
+    // Black Duck fields will be common as long as it is the only provider
+    private final Long blackDuckGlobalConfigId;
+    private final boolean filterByProject;
+    private final String projectNamePattern;
+    private final List<String> notificationTypes;
+    private final List<String> projectFilterProjectNames;
+    private final List<String> policyFilterPolicyNames;
+    private final List<String> vulnerabilityFilterSeverityNames;
+
+    private final DistributionJobDetailsModel distributionJobDetails;
+
+    public static DistributionJobModelBuilder builder() {
+        return new DistributionJobModelBuilder();
+    }
+
+    protected DistributionJobModel(
+        UUID jobId,
+        boolean enabled,
+        String name,
+        FrequencyType distributionFrequency,
+        ProcessingType processingType,
+        String channelDescriptorName,
+        OffsetDateTime createdAt,
+        OffsetDateTime lastUpdated,
+        Long blackDuckGlobalConfigId,
+        boolean filterByProject,
+        String projectNamePattern,
+        List<String> notificationTypes,
+        List<String> projectFilterProjectNames,
+        List<String> policyFilterPolicyNames,
+        List<String> vulnerabilityFilterSeverityNames,
+        DistributionJobDetailsModel distributionJobDetails
+    ) {
+        this.jobId = jobId;
+        this.enabled = enabled;
+        this.name = name;
+        this.distributionFrequency = distributionFrequency;
+        this.processingType = processingType;
+        this.channelDescriptorName = channelDescriptorName;
+        this.createdAt = createdAt;
+        this.lastUpdated = lastUpdated;
+        this.blackDuckGlobalConfigId = blackDuckGlobalConfigId;
+        this.filterByProject = filterByProject;
+        this.projectNamePattern = projectNamePattern;
+        this.notificationTypes = notificationTypes;
+        this.projectFilterProjectNames = projectFilterProjectNames;
+        this.policyFilterPolicyNames = policyFilterPolicyNames;
+        this.vulnerabilityFilterSeverityNames = vulnerabilityFilterSeverityNames;
+        this.distributionJobDetails = distributionJobDetails;
+    }
+
+    @Nullable
```

Comment:
When do we prefer @Nullable on methods vs constructor param?

Reply:
I think I meant to switch to Optional. I'll double check.